File: Contract/Dto/ArticleDto.cs
using System.Runtime.Serialization;

namespace DroneNews.Contract.Dto;

public class ArticleDto
{
    public int Id { get; set; }

    public string Title { get; set; }

    public string Description { get; set; }

    public string Content { get; set; }

    public string OriginalUrl { get; set; }

    public string? ImageUrl { get; set; }

    public DateTime PublishedAt { get; set; }

    public string? Author { get; set; }

    public SourceDto? Source { get; set; }
}

File: dronenews.client/src/server/model/article.dto.ts
import type {SourceDto} from "./source.dto.ts";

export interface ArticleDto {
    id: number;
    title: string;
    description: string;
    content: string;
    originalUrl: string;
    imageUrl?: string;
    publishedAt: Date;
    author?: string;
    source?: SourceDto;
}

